summoner-cli/src/Summoner/Template/GitHub.hs

```diff
@@ -121,49 +121,113 @@ gitHubFiles Settings{..} = concat
         , "    branches: [master]"
         , ""
         , "jobs:"
-        , "  build:"
-        , "    name: ghc ${{ matrix.ghc }}"
-        , "    runs-on: ubuntu-16.04"
+        ]
+        <> memptyIfFalse settingsCabal ghActionsCabal
+        <> memptyIfFalse settingsStack ghActionsStack
+
+    ghActionsCabal :: [Text]
+    ghActionsCabal =
+        [ "  cabal:"
+        , "    name: ${{ matrix.os }} / ghc ${{ matrix.ghc }}"
+        , "    runs-on: ${{ matrix.os }}"
         , "    strategy:"
         , "      matrix:"
+        , "        os: [ubuntu-latest, macOS-latest, windows-latest]"
         , "        cabal: [" <> quote defaultCabal <> "]"
         , "        ghc:"
         ]
         <> map (indent 10 <>) ghActionsVersions
         <>
+        [ "        exclude:"]
+        <> map (indent 10 <>) ghActionsExcludes
+        <>
         [ ""
         , "    steps:"
         , "    - uses: actions/checkout@v2"
         , "      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'"
         , ""
-        , "    - uses: actions/setup-haskell@v1"
+        , "    - uses: actions/setup-haskell@v1.1.1"
```

Comment:
Since we are implementing this, let's also use the latest versions of all actions at the moment. It probably won't be that easy to update, but for now we can use the latest versions. For `setup-haskell` the latest version if `1.1.3`. And, I guess, we can move all such versions into variables to make the easier to update.
```suggestion
        , "    - uses: actions/setup-haskell@v${defaultSetupHaskellVersion}"
```

Reply:
I will use the newest versions of actions and extract the versions into variables to make maintenance easier.

Reply:
Done.